packages/nouns-contracts/contracts/Rewards.sol

```diff
@@ -320,6 +320,42 @@ contract Rewards is NounsClientToken, UUPSUpgradeable {
         return balance;
     }
 
+    /**
+     * @notice Returns the clientIds that is needed to be passed as a parameter to updateRewardsForProposalWritingAndVoting
+     * @dev This is not meant to be called onchain because it may be very gas intensive.
+     */
+    function getVotingClientIds(uint32 lastProposalId) public view returns (uint32[] memory) {
+        uint256 numClientIds = nextTokenId();
+        uint32[] memory allClientIds = new uint32[](numClientIds);
```

Comment:
there's a risk someone might create so many client IDs that this function becomes very gas-intensive such that Etherscan can't run it.

I think it's not very interesting to mitigate this risk right now, but thought it good to discuss it here explicitly.

wdyt?

Reply:
yes it's possible. I'm not so concerned since there's always the option of using an indexer to figure out the client ids that voted for a proposal